engines/sg.c

```diff
@@ -3,6 +3,39 @@
  *
  * IO engine that uses the Linux SG v3 interface to talk to SCSI devices
  *
+ * This ioengine can operate in two modes:
+ *	sync	with block devices (/dev/sdX) or
+ *		with character devices (/dev/sgY) with direct=1 or sync=1
+ *	async	with character devices with direct=0 and sync=0
+ *
+ * What value does queue() return for the different cases?
+ *				queue() return value
+ * In sync mode:
+ *  /dev/sdX		RWT	FIO_Q_COMPLETED
+ *  /dev/sgY		RWT	FIO_Q_COMPLETED
+ *   with direct=1 or sync=1
+ *
+ * In async mode:
+ *  /dev/sgY		RWT	FIO_Q_QUEUED
+ *   direct=0 and sync=0
+ *
+ * Because FIO_SYNCIO is set for this ioengine td_io_queue() will fill in issue_time *before* each IO is sent to queue()
+ *
+ * Where are the IO counting functions called for the different cases?
+ * In sync mode:		io_u_mark_depth()		io_u_mark_submit/comp()		issue_time
+ *  /dev/sdX:		RWT	called in td_io_queue()		called in td_io_queue()		set in td_io_queue()
+ *  commit==NULL
+ *
+ *  /dev/sgY with direct=1 or sync=1:
+ *  commit does nothing	RWT	called in td_io_queue()		called in queue()		set in td_io_queue()
+ *
+ *
+ * In async mode:		read and write operations are submitted in queue()
+ *			RW	called in queue()		submit called in queue()	set in td_io_queue()
+ *
+ *				trim operations are queued in queue() and submitted in commit()
+ *			T	called in commit()		submit called in commit()	set in commit()
+ *
```

Comment:
I like the detailed explanations, but please keep it within 80 chars. It's unreadable in my term.